lib/https.c

```diff
@@ -351,9 +351,9 @@ _establish_connection(struct https_request * const req,
 
 /* Provide implementations for HMAC_CTX_new and HMAC_CTX_free when
  * building for OpenSSL versions older than 1.1.0
- * or any version of LibreSSL.
+ * or LibreSSL versions older than 2.7.0
  */
-#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || LIBRESSL_VERSION_NUMBER < 0x2070000fL
```

Comment:
I think we need to be slightly more robust and first run
`defined(LIBRESSL_VERSION_NUMBER)` before checking it against 2.7.0

I ran a quick test on a machine without libressl and saw that
`#if LIBRESSL_VERSION_NUMBER < 0x2070000fL` was evaluating to true. I think this would cause systems with openssl 1.1.0 to fail like the libressl ones are failing now.

So in the end it could look like
`if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x2070000fL`

Reply:
Thanks Xander. Good call on the check.

I pushed the updated commit.

Unfortunately travis fails.
Fortunately (for my change), master fails at the same place as seen at https://github.com/duosecurity/duo_unix/pull/132

Reply:
Oh bummer! thanks finding out a broken/flakey build. I'm gonna try to get master fixed and then we'll pull in this change. I may also just try a quick rebuild of this PR to see if it's just flakey.

Reply:
sounds good. let me know if you'd like me to rebase when master is fixed.